src/Angor/Client/Pages/Signatures.razor

```diff
@@ -408,103 +423,118 @@
         }
+
+        StateHasChanged();
     }
 
-    private async Task ApproveSignature(SignatureRequest signature)
+    private async Task ApproveAllSignatures()
     {
-        var words = await passwordComponent.GetWalletAsync();
-
-        var operationResult = await PerformSignatureApproval(signature, words);
-
-        if (!operationResult.Success)
+        if (passwordComponent.HasPassword())
         {
-            notificationComponent.ShowErrorMessage(operationResult.Message);
+            await ProcessSignatures();
         }
         else
         {
-            StateHasChanged();
+            passwordComponent.ShowPassword(async () =>
+            {
+                await ProcessSignatures();
+            });
         }
     }
 
-    private async Task ApproveAllSignatures()
+    private async Task ProcessSignatures()
     {
         isLoading = true;
         StateHasChanged();
 
         try
         {
-            if (passwordComponent.HasPassword())
-            {
-                await ProcessSignatures();
-            }
-            else
+            var pendingSignatures = signaturesRequests.Where(s => s.TransactionHex != null && s.AmountToInvest != null && s.TimeApproved == null).ToList();
+            numOfSignatureToSign = pendingSignatures.Count;
+            numOfSignaturesSigned = 0;
+        
+            var words = await passwordComponent.GetWalletAsync();
+        
+            foreach (var signature in pendingSignatures)
             {
-                passwordComponent.ShowPassword(async () =>
-                {
-                    await ProcessSignatures();
-                });
+                await PerformSignatureApproval(signature, words);
+                numOfSignaturesSigned++;
+                StateHasChanged();
             }
         }
         catch (Exception e)
         {
-            Logger.LogError("Error approving signatures: {ErrorMessage}", e.Message, e);
+            Logger.LogError(e, e.Message);
+            notificationComponent.ShowErrorMessage(e.Message);
         }
         finally
         {
             isLoading = false;
             passwordComponent.ClearPassword();
-            StateHasChanged();
         }
+        
+        StateHasChanged();
     }
 
-    private async Task ProcessSignatures()
+    private async Task<OperationResult> PerformSignatureApproval(SignatureRequest signature, WalletWords words)
     {
-        var pendingSignatures = signaturesRequests.Where(s => s.TransactionHex != null && s.AmountToInvest != null && s.TimeApproved == null).ToList();
-        numOfSignatureToSign = pendingSignatures.Count;
-        numOfSignaturesSigned = 0;
-        
-        var words = await passwordComponent.GetWalletAsync();
-        
-        foreach (var signature in pendingSignatures)
+        try
         {
-            await PerformSignatureApproval(signature, words);
-            numOfSignaturesSigned++;
-            StateHasChanged();
+            if (words == null)
```

Comment:
i'm noticing it now, words can't be null from 
`passwordComponent.GetWalletAsync();` right? thats why we removed the check before.
so we can remove the check here also

Reply:
yes correct I will remove it